src/Tracer.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Exercism\PhpTestRunner;
+
+use Exercism\PhpTestRunner\Result;
+use PHPUnit\Event\Code\TestMethod;
+use PHPUnit\Event\Event;
+use PHPUnit\Event\Test\BeforeFirstTestMethodErrored;
+use PHPUnit\Event\Test\Errored;
+use PHPUnit\Event\Test\Failed;
+use PHPUnit\Event\Test\Passed;
+use PHPUnit\Event\Test\PrintedUnexpectedOutput;
+use PHPUnit\Event\TestRunner\Finished;
+use PHPUnit\Event\Tracer\Tracer as TracerInterface;
+use ReflectionClass;
+
+final class Tracer implements TracerInterface
+{
+    private array $result = [
+        'version' => 3,
+        'status' => 'pass',
+        'tests' => [],
```

Comment:
I think there should be typing here: `Result[]`

Reply:
How to add that type? I have no idea :man_shrugging:

Reply:
You can declare array shape like this:

```php
/** @var array{version: int, status: string, tests: list<Result>} $result  */
private array $result = ...;
```